feat: add `lapack/base/dlapy3`

aayush0325 opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   adds `lapack/base/dlapy3`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

aayush0325 committed:
feat: add lapack/base/dlapy3

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
test: add tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (changes requested):
Same general comments as `dlapy2`.

aayush0325 committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

aayush0325 committed:
chore: code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/docs/repl.txt:
```diff
@@ -0,0 +1,30 @@
+
+{{alias}}( x, y, z )
+    Computes `sqrt( x^2 + y^2 + z^2 )` in a manner which doesn't cause
```
```suggestion
    Computes `sqrt(x^2 + y^2 + z^2)` in a manner which doesn't cause
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/docs/repl.txt:
```diff
@@ -0,0 +1,30 @@
+    Returns
+    -------
+    out: number
+        Square root of sum of squares of all input numbers.
```
```suggestion
        Square root of sum of squares.
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/docs/types/test.ts:
```diff
@@ -0,0 +1,64 @@
+	dlapy3( 8, 2, 10 ); // $ExpectType number
+}
+
+// The compiler throws an error if the function is provided values other than two numbers...
```
```suggestion
// The compiler throws an error if the function is provided values other than three numbers...
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/docs/types/index.d.ts:
```diff
@@ -0,0 +1,42 @@
+// TypeScript Version: 4.1
+
+/**
+* LAPACK routine to calculate `sqrt( x^2 + y^2 + z^2 )` in a manner which doesn't cause unnecessary overflow.
```
```suggestion
* LAPACK routine to calculate `sqrt(x^2 + y^2 + z^2)` in a manner which doesn't cause unnecessary overflow.
```

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/lib/main.js:
```diff
@@ -0,0 +1,69 @@
+	if ( w === 0.0 || w > hugeval ) {
+		return xabs + yabs + zabs;
+	}
+
```
```suggestion
```

kgryte committed:
style: remove empty line

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
docs: update comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
style: remove spaces

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/lib/main.js:
```diff
@@ -0,0 +1,68 @@
+/**
+* Computes `sqrt(x^2 + y^2 + z^2)` in a manner which doesn't cause unnecessary overflow.
+*
+* @private
```
```suggestion
```

kgryte committed:
docs: remove annotation

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/test/test.js:
```diff
@@ -0,0 +1,94 @@
+	t.strictEqual( h, 0.0, 'returns 0' );
+
+	h = dlapy3( 1.0e-200, 1.0e-200, 1.0e-200 );
+	t.strictEqual( h, 1.7320508075688772e-200, 'avoids underflow' );
```
```suggestion
	t.strictEqual( h, 1.7320508075688772e+308, 'returns expected value' );

	t.end();
});

tape( 'the function avoids underflow', function test( t ) {
	var h;

	h = sqrt( pow( 1.0e-200, 2 ) + pow( 1e-200, 2 ) + pow( 1e-200, 2 ) );
	t.strictEqual( h, 0.0, 'returns expected value' );

	h = dlapy3( 1.0e-200, 1.0e-200, 1.0e-200 );
	t.strictEqual( h, 1.7320508075688772e-200, 'returns expected value' );
```

kgryte committed:
test: update messages

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
test: add `NaN` tests

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/lapack/base/dlapy3/test/test.js:
```diff
@@ -0,0 +1,122 @@
+	t.end();
+});
+
+tape( 'the function computes the euclidian norm', function test( t ) {
```
```suggestion
tape( 'the function computes the Euclidian norm', function test( t ) {
```

kgryte committed:
test: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after clean-up.
